### PR TITLE
Clean up optimization to skip layers for opacity elements

### PR DESCRIPTION
**Pull request by tronical**

### CODE REVIEW

File: internal/core/items.rs [hunk 1]
```diff
@@ -1164,22 +1152,7 @@ impl Opacity {
...
-                if children_count == 1 =>
-            {
-                children_index as usize
-            }
-            _ => return true, // Dynamic tree or multiple children -> need layer
-        };
-        let target_item = ItemRc::new(component_rc.clone(), target_item_index);
-        // any children? Then we need a layer
-        target_item.children_count() != Some(0)
+        self_rc.first_child().and_then(|target_item| target_item.first_child()).is_some()
```
Comment (ogoffart) [inline internal/core/items.rs:1155]: Iirc, we also said we would need to check that the opacity element only have one child

Comment (tronical) [reply internal/core/items.rs]: I'm not sure I understand. I tried to make the code equivalent. I realize the above isn't, but I pushed what I think is a fix.

Comment (ogoffart) [reply internal/core/items.rs]: If the Opacity item has two or more children (potentially overlapping)  we need to create a layer, which this code doesn't do as it only check the first children and doesn't check if it has a sibling.

If there is no children at all, no need to create a layer.

Comment (tronical) [reply internal/core/items.rs]: I see now what you mean (I think ;-). Thanks for spotting.

**Review by ogoffart — APPROVED** [review]

**Pull request merged**